graylog2-server/src/main/java/org/graylog2/rest/resources/system/ConfigurationResource.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.graylog2.rest.resources.system;
+
+import com.codahale.metrics.annotation.Timed;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.graylog2.Configuration;
+import org.graylog2.configuration.ElasticsearchConfiguration;
+import org.graylog2.rest.models.system.configuration.ConfigurationList;
+import org.graylog2.rest.models.system.configuration.ConfigurationVariable;
+import org.graylog2.shared.rest.resources.RestResource;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiresAuthentication
+@Api(value = "System/Configuration", description = "Read-only access to configuration variables")
+@Path("/system/configuration")
+public class ConfigurationResource extends RestResource  {
+
+    private final Configuration configuration;
+    private final ElasticsearchConfiguration esConfiguration;
+
+    @Inject
+    public ConfigurationResource(Configuration configuration, ElasticsearchConfiguration esConfiguration) {
+        this.configuration = configuration;
+        this.esConfiguration = esConfiguration;
+    }
+
+    /*
+     * This call is manually building a map of variables to return because we need to guarantee never to return any
+     * sensitive variables like passwords etc. - See this as a whitelist approach.
+     */
+    @GET
+    @Timed
+    @ApiOperation(value = "Get all reported configuration variables and their values")
+    public ConfigurationList getAll() {
```

Comment:
What about adding jackson annotations (JsonProperty/JsonIgnore, whichever way around we prefer) to the Configuration class and return it in the REST resource instead of using these convoluted value objects and the very verbose way of creating them?